proofs/alf/cvc5/Cvc5.smt3

```diff
@@ -78,13 +84,27 @@
       ((run_evaluate (div x y))          (alf.zdiv (run_evaluate x) (run_evaluate y)))
       ((run_evaluate (div_total x y))    (let ((d (run_evaluate y)))
                                            (alf.ite (alf.is_eq d 0) 0 (alf.zdiv (run_evaluate x) d))))
+      ((run_evaluate (mod x y))          (alf.zmod (run_evaluate x) (run_evaluate y)))
+      ((run_evaluate (to_real x))        (alf.to_q (run_evaluate x)))
+      ((run_evaluate (to_int x))         (alf.to_z (run_evaluate x)))
+      ((run_evaluate (is_int x))         (alf.is_eq (alf.to_q (alf.to_z x)) (alf.to_q x)))
+
 
       ; strings
       ((run_evaluate (str.++ xs yss))    (alf.concat (run_evaluate xs) (run_evaluate yss)))
       ((run_evaluate (str.len x))        (alf.len (run_evaluate x)))
       ((run_evaluate (str.substr x n m)) (let ((r (run_evaluate n)))
                                             (alf.extract (run_evaluate x) r (alf.add r (run_evaluate m) -1))))
       ((run_evaluate (str.contains x y)) (alf.not (alf.is_neg (alf.find (run_evaluate x) (run_evaluate y)))))
+      ((run_evaluate (str.replace x y z))(let ((ex (run_evaluate x)))
+                                         (let ((ey (run_evaluate y)))
+                                         (let ((i (alf.find ex ey)))
+                                            (alf.ite (alf.or (alf.is_eq i 0) (alf.is_neg (alf.neg i)))
+                                              (alf.concat (alf.extract ex 0 (alf.add i -1)) (run_evaluate z) (alf.extract ex (alf.add i (alf.len ey)) (alf.len ex)))
+                                              ex)))))
```

Comment:
Not sure about the number of parentheses

```suggestion
                                            (alf.ite (alf.is_neg i)
                                              ex
                                              (alf.concat (alf.extract ex 0 (alf.add i -1)) (run_evaluate z) (alf.extract ex (alf.add i (alf.len ey)) (alf.len ex))))))))
```

Also, shouldn't the last index be `(alf.add (alf.len ex) -1)`?

Reply:
It doesn't matter, it just has to be enough characters to go pass the endpoint of the string, so `(alf.len ex)` suffices and is easier to compute.